x-pack/test/functional/apps/api_keys/home_page.ts

```diff
@@ -13,11 +13,26 @@ export default ({ getPageObjects, getService }: FtrProviderContext) => {
   const log = getService('log');
   const security = getService('security');
   const testSubjects = getService('testSubjects');
+  const es = getService('es');
   const find = getService('find');
   const browser = getService('browser');
 
+  async function clearAllApiKeys() {
+    const existingKeys = await es.security.queryApiKeys();
+    if (existingKeys.count > 0) {
+      await (
+        await es.security.queryApiKeys()
+      ).api_keys.map(async (key) => {
+        await es.security.invalidateApiKey({ ids: [key.id] });
+      });
+    } else {
+      log.debug('No API keys to delete.');
+    }
+  }
```

Comment:
Afaik you [need](https://medium.com/sliit-foss/js-async-await-in-array-methods-9142a35c6d6f) `Promise.all` to use async operation in `Array.map`. Also it is quite redundant to fetch API Keys twice.

```suggestion
  async function clearAllApiKeys() {
    const keys = await es.security.queryApiKeys();
    if (keys.count > 0) {
      await Promise.all(keys.api_keys.map((key) => es.security.invalidateApiKey({ ids: [key.id] })));
    } else {
      log.debug('No API keys to delete.');
    }
  }
```

Reply:
Fixed.